core/src/mindustry/entities/bullet/BulletType.java

```diff
@@ -700,11 +700,19 @@ public ContentType getContentType(){
         return create(owner, team, x, y, angle, damage, velocityScl, lifetimeScl, data, null);
     }
 
+    public @Nullable Bullet create(@Nullable Entityc owner, @Nullable Entityc shooter, Team team, float x, float y, float angle, float damage, float velocityScl, float lifetimeScl, Object data){
+        return create(owner, shooter, team, x, y, angle, damage, velocityScl, lifetimeScl, data, null, -1, -1);
+    }
+
     public @Nullable Bullet create(@Nullable Entityc owner, Team team, float x, float y, float angle, float damage, float velocityScl, float lifetimeScl, Object data, @Nullable Mover mover){
         return create(owner, team, x, y, angle, damage, velocityScl, lifetimeScl, data, mover, -1f, -1f);
     }
 
     public @Nullable Bullet create(@Nullable Entityc owner, Team team, float x, float y, float angle, float damage, float velocityScl, float lifetimeScl, Object data, @Nullable Mover mover, float aimX, float aimY){
+        return create(owner, owner, team, x, y, angle, damage, velocityScl, lifetimeScl, data, mover, aimX, aimY);
+    }
+
+    public @Nullable Bullet create(@Nullable Entityc owner, @Nullable Entityc shooter, Team team, float x, float y, float angle, float damage, float velocityScl, float lifetimeScl, Object data, @Nullable Mover mover, float aimX, float aimY){
```

Comment:
Instead of adding overloads, couldn't you do something like checking if the shooter has an owner, and if so, assigning it to the shooter's owner?

Reply:
What if the missile fires a bullet which then frags into more missiles?
I'm just trying to account for passing through any amount of bullet/missile/frag shenanigans.

Reply:
> What if the missile fires a bullet which then frags into more missiles?

- Missile shoooter is unit
- Frags are assigned missile's shooter
- Missiles from frags are assigned to frag's shooter, which is still the original shooter
- and so on